src/ucp/core/ucp_mm.c

```diff
@@ -461,6 +465,17 @@ static ucs_status_t ucp_memh_register(ucp_context_h context, ucp_mem_h memh,
     memh->md_map |= md_map_registered;
     status        = UCS_OK;
 
+    if ((context->config.ext.reg_nb_mem_types & UCS_BIT(mem_type)) &&
+        (md_map_registered == 0)) {
+        ucs_log(err_level,
+                "Available MDs cannot perform non-blocking registration "
+                "for %s memory type", ucs_memory_type_names[mem_type]);
+
+        if (!(uct_flags & UCT_MD_MEM_FLAG_HIDE_ERRORS)) {
+            status = UCS_ERR_UNSUPPORTED;
+        }
+    }
```

Comment:
IMO don't need to consider this an error. it's ok we don't register at all

Reply:
We have assertion that will fail if we consider this case not an error. https://github.com/openucx/ucx/blob/master/src/ucp/core/ucp_mm.c#L849

Should we remove this assertion too?

BTW what is the case when not-registering memory region is OK? As I understand it will just fail during the later calls, am I correct?

Reply:
in theory it will fall-back to some (possibly non-optimal) transport